### DISCUSSION TITLE
Multiprocessing

### DISCUSSION THREAD

**Discussion by didieransys** [ansys/pydpf-core#1707] [General]
We have an application that needs to extract data using DPF in a loop. We want to take advantage of parallelism and we need to have some insight about that is recommended or even working to parallelize DPF operations.
Should we launch multiple DPF servers in order to distribute the workload? Does one DPF server handle concurrent operations? Would multiple DPF server share a common backend? Would that even work? Is the DPF API thread safe? 
I have not been able to locate information about all that, but I might have missed it. Any hint will be appreciated.

**Comment by rlagha** [comment]
Hi @didieransys ,
both can work, distributing the calculations with one server or several ones. Lets have a meeting on it, and can fill this discussion with more details.

**Comment by xcast-ppi** [comment]
Hi @rlagha, 
I'm doing similar try as didieransys. 

I've found that while spawning parallel workers they need a server to be built in each task independently and build a "model" from a RST for each one of them. But unfortunately at the expense of an Ansys license. Otherwise, if I try to build one server in the beginning and pass it to the tasks as arguments, I get errors that some object cannot be pickled.

Is there a way to reuse the same server and yet launch several parallel tasks that need a model from a RST file independently?
Or, instead, to reuse the license requested?

Thanks
